models/reference.py

```diff
@@ -219,7 +219,7 @@ def load_upstream_products(self, session=None):
 
             sources = session.scalars(
                 sa.select(SourceList).where(
-                    SourceList.image_id == self.image_id,
+                    SourceList.image_id == self.image.id,
```

Comment:
Why not leave this as is?  I guess if you're doing eager loading, then `self.image` will always be loaded, but it seems simpler and more straightforward to just use the `image_id` property.

Reply:
I don't remember but there was test where that didn't work. I am not sure what the situation was...